app/assets/javascripts/discourse/app/components/bookmark-list.js

```diff
@@ -84,17 +86,20 @@ export default Component.extend(Scrolling, {
 
   @action
   editBookmark(bookmark) {
-    openBookmarkModal(bookmark, {
-      onAfterSave: (savedData) => {
-        this.appEvents.trigger(
-          "bookmarks:changed",
-          savedData,
-          bookmark.attachedTo()
-        );
-        this.reload();
-      },
-      onAfterDelete: () => {
-        this.reload();
+    this.modal.show(BookmarkModal, {
+      model: {
+        bookmark: new BookmarkFormData(bookmark),
```

Comment:
This BookmarkFormData thing is not super ideal but I didn't want to go into the rabbit hole of changing Bookmark model to not use RestModel at the same time as well.